File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.py
class Solution:
    def __init__(self):
        self.vowels = set(['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'])
    def halvesAreAlike(self, s: str) -> bool:
        count = 0
        for i in range(len(s)//2):
            count += 1 if s[i] in self.vowels else 0
        for i in range(len(s)//2):
            count -= 1 if s[-(i+1)] in self.vowels else 0
            if count < 0:
                return False
        return not(count)

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.rs
use std::collections::HashSet;

impl Solution {
    pub fn halves_are_alike(s: String) -> bool {
        let vowels: HashSet<char> = HashSet::from(['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']);
        let s_vec: Vec<char> = s.chars().collect();
        let mut count: i32 = 0;
        let n: i32 = s.len() as i32;
        for i in 0..(n/2) {
            if (vowels.contains(&s_vec[i as usize])) {
                count += 1
            }
        }
        for i in 0..(n/2) {
            if (vowels.contains(&s_vec[(n-i-1) as usize])) {
                count -= 1;
                if (count < 0) {
                    return false;
                }
            }
        }
        return (count == 0);
    }
}